nssp/delphi_nssp/patch.py

```diff
@@ -0,0 +1,83 @@
+"""
+This module is used for patching data in the delphi_nssp package.
+
+To use this module, you need to specify the range of issue dates in params.json, like so:
+
+{
+  "common": {
+    ...
+  },
+  "validation": {
+    ...
+  },
+  "patch": {
```

Comment:
As I got feedback from Nat from my google-symptom patch, I realized that if we don't remove the patch parameter, wonky things could happen. So for me, I decided to add another parameter in common called custom_run as a safety check
line 58-60
```
    # safety check for patch parameters exists in file, but not running custom runs/patches
    custom_run_flag = False if not params["common"].get("custom_run") else params["common"].get("custom_run", False)
 ```
 I thought it would be better have a new flag rather than to remove back and forth. 
 
 second note for me too; we should have a section for params.json in the DEV guide of the readme @nmdefries thought? oh and also add patch in the Step 2, development sub section where we list out the core modules that an indicator has

Reply:
There's an issue for [adding patching info](https://github.com/cmu-delphi/covidcast-indicators/issues/1993) to the dev guide. Just added an [issue for adding params info](https://github.com/cmu-delphi/covidcast-indicators/issues/2002) to the dev guide, too.

Reply:
@aysim319 so for the custom run flag, I wanna make sure the behavior makes sense:
| custom_run_flag  | "patch" and others exist in params | -> run patch?|
| ------------- | ------------- |------------- |
| true  | true | true |
| false  | false  | false |
| true  | false | error?/false |
| false  | true | false |
Is this approach reasonable?

Reply:
1st, 2nd, last is correct. I didn't realize at least for nssp, it doesn't have a start_date, end_date parameter like some of the other ones. 

I think having a warning saying that this flag is on, but no custom date to refer and don't run patch, but just treat it as regular run is probably the way to go. Because we don't want to exit and throw an error when it's not something that could break the program.

Reply:
Putting patch info into the params file has always bothered me.  It wastes time by complicating the patching process (ie: edit params file, run, remove edits from file), and it is fragile and dangerous (what if someone forgets to remove the file changes?), and probably other downsides that i cant think of right now.  Patches are pretty much always run by hand, so IMNSHO the patch-specific parameters should be specified by commandline arguments.  It will make things easier and more portable.

Reply:
If you want the patching process to use parameters from a file instead of CLI (there are good reasons for it), i would recommend creating a separate runner that does something like read a `patch.params.json` file and combining that with the regular `params.json` before passing the result to the regular runner.

That being said, i havent had a chance to review this PR yet, so im not up to speed on what youre doing here...  I was directed here via a comment in #2002 .

Reply:
@melange396 made changes to run as a cli for #1999 is this something you were looking for?

Reply:
@melange396 I support the extra `patch.params.json` path over the cli path. Last year I when I wrote the early versions of these patching scripts, I actually tried adding and using cli. I ended up going back to params for a few reasons:
- The number and length of input becomes a bit too long and not readable, specifically when long dir/paths are involved.
- When running patches, I notice that I most times I do open up params.json to make other configurations specific to the need of the patch (geos, credentials, archive, etc.). So the process would involve editing params.json anyway, and now I gotta both edit params.json and think of cli configuration.
- Which is why I only run patches on staging (and then put results on prod for acquisition), hence the accidental prod run using patch params won't happen.

Reply:
I like the idea of having a separate `patch.params.json` file. I like having a saved file available that already has the right format defined (e.g. so you can see that the date needs to be YYYYMMDD, not another format). It decreases time spent chasing down docs and command line arg names.

Reply:
> most times I do open up params.json to make other configurations specific to the need of the patch

Hm, if you have to do this a lot it could be inconvenient to have a separate `patch.params.json`, since you'd end up needing to modify two separate files instead of one.

Reply:
yea that's true lmao. I think @aysim319 suggestion about the flag put us in a good direction for now in terms of preventing accidental patch run.

Reply:
So... are we keeping with having a custom_run flag and having patch params within the original params.json or having a new one for patch.params.json?

Reply:
For now I'll add the custom run flag for this pr. I think if we want to do patch.params.json it should be a separate effort anyway.